app/service/form_data_test.go

```diff
@@ -0,0 +1,210 @@
+package service
+
+import (
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestFormData_ParseJSONRequestData(t *testing.T) {
+	tests := []struct {
+		name                string
+		definitionStructure interface{}
+		json                string
+		wantErr             string
+		wantFieldErrors     FieldErrors
+	}{
+		{
+			"simple",
+			&struct {
+				ID   int64  `json:"id"`
+				Name string `json:"name"`
+			}{},
+			`{"id":123, "name":"John"}`,
+			"",
+			nil,
+		},
+		{
+			"invalid JSON",
+			&struct{}{},
+			`{id:123, name:"John"}`,
+			"invalid character 'i' looking for beginning of object key string",
+			nil,
+		},
+		{
+			"wrong value for a field",
+			&struct {
+				ID int64 `json:"id"`
+			}{},
+			`{"id":"123"}`,
+			"invalid input data",
+			FieldErrors{"id": {"expected type 'int64', got unconvertible type 'string'"}},
+		},
+		{
+			"unexpected field",
+			&struct {
+				ID int64 `json:"id"`
+			}{},
+			`{"my_id":"123"}`,
+			"invalid input data",
+			FieldErrors{"my_id": {"unexpected field"}},
+		},
+		{
+			"field ignored by json",
+			&struct {
+				Name string `json:"-" gorm:"column:sName"`
+			}{},
+			`{"Name":"test"}`,
+			"invalid input data",
+			FieldErrors{"Name": {"unexpected field"}},
+		},
+		{
+			"decoder error for a field",
+			&struct {
+				Time time.Time `json:"time"`
+			}{},
+			`{"time":"123"}`,
+			"invalid input data",
+			FieldErrors{"time": {"decoding error: parsing time \"123\" as \"2006-01-02T15:04:05Z07:00\": cannot parse \"123\" as \"2006\""}},
+		},
+		{
+			"multiple errors",
+			&struct {
+				ID int64 `json:"id" valid:"required"`
```

Comment:
Failing on `required` when `0` is given (which is a value) seems something very "Go-ish" that is quite wrong. Still the same need for a distinction between the default Go value <> no value given <> null given!

Reply:
We must not use 'required', it is here just for testing purposes